samples/data-sealing/common/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Use the edger8r to generate C bindings from the EDL file.
+add_custom_command(OUTPUT datasealing_t.h datasealing_t.c
+  COMMAND openenclave::oeedger8r --trusted ${CMAKE_SOURCE_DIR}/datasealing.edl)
+
+# Create a library common to each of our three enclaves.
+add_library(common STATIC dispatcher.cpp keys.cpp ${CMAKE_CURRENT_BINARY_DIR}/datasealing_t.c)
+target_link_libraries(common PUBLIC
+  # `liboecore`, a dependency of `liboeenclave`, requires the ecalls
+  # function table. Because the libraries linking `libcommon` do not
+  # directly require this symbol, the linker skips the object in
+  # `libcommon` which defines them. So we manually require it.
+  #
+  # Alternatively we could use a CMake OBJECT library, but that
+  # requires a newish version of CMake.
+  -Wl,--require-defined=__oe_ecalls_table
```

Comment:
Would everyone implementing a sample with cmake need to force the linker to include __oe_ecalls_table?

Reply:
No, this is only if you use a common static library like done here. Perhaps I could make the comment more clear?

Reply:
Sure. This is really complicated, though :(.

Reply:
I know! I told @soccerGB he gave me the worst sample. This exhibits the creation of a static library of common enclave code, linked into multiple enclaves, which is an edge case which requires this complicated-ness (though it's not that different from similar scenarios that aren't in enclaves) and is valuable to have as an example.

@aserdean is going to follow up and implement CMake package based builds for the rest of the samples, which should be much simpler and therefore easier to follow as well. But he's blocked until this goes in.